js/scripts/util.js

```diff
@@ -172,10 +181,58 @@ const MUSIC = {
     LAVA_TENSE: { path: "./music/lava_tense.mp3", volume: 0.1 }, // might be timed task OR boss music
 }
 
+/**
+ * Gets an array of nearby entities from the midground.
+ * The entities must have a bounding box.
+ * 
+ * NOTE: I tried to optimize this as much as possible, but it still looks expensive with all the distance calculations.
```

Comment:
This won't be too much more expensive than collision checking, if at all.

Reply:
Good to know. Thanks!